toolchain/check/handle_class.cpp

```diff
@@ -651,10 +658,37 @@ static auto CheckCompleteClassType(Context& context, Parse::NodeId node_id,
     return CheckCompleteAdapterClassType(context, node_id, class_id, fields_id);
   }
 
+  bool defining_vtable_ptr = class_info.is_dynamic;
+  if (class_info.base_id.is_valid()) {
+    auto base_info = context.insts().GetAs<SemIR::BaseDecl>(class_info.base_id);
+    if (auto* base_class_info = TryGetAsClass(context, base_info.base_type_id);
+        base_class_info && base_class_info->is_dynamic) {
+      defining_vtable_ptr = false;
+    }
+  }
+
+  llvm::SmallVector<SemIR::InstId> object_rep_fields;
+  auto fields = context.inst_blocks().Get(fields_id);
+  if (defining_vtable_ptr) {
+    if (!fields.empty()) {
+      auto field =
+          context.insts().GetAs<SemIR::StructTypeField>(fields.front());
+      object_rep_fields.push_back(
+          context.AddInstInNoBlock<SemIR::StructTypeField>(
+              Parse::NodeId::Invalid,
+              {.name_id = SemIR::NameId::Vptr,
+               .field_type_id = context.GetPointerType(field.field_type_id)}));
+    }
+  }
+
+  object_rep_fields.append(fields.begin(), fields.end());
```

Comment:
Perhaps...
```suggestion
    object_rep_fields.append(fields.begin(), fields.end());
  }
```
given that you don't use `object_rep_fields` in the `!defining_vtable_ptr` case.